docs/_tutorials/reference.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Reference
+order: 1
+slug: reference
+lang: en
+layout: fullpage
+permalink: /reference
+---
+# App interface and configuration
+
+<div class="section-content">
+This guide is intended to detail the Bolt interface–including listeners and their arguments, initialization options, and errors. It may be helpful to first go through the ⚡️[Getting Started guide](/getting-started) to learn the basics of building Bolt for JavaScript apps.
+</div> 
+
+- [Listener middleware](#listener-middleware)
+  - [Methods](#methods)
+  - [Function arguments](#listener-function-arguments)
+- [Initialization options](#initialization-options)
+  - [Receiver options](#receiver-options)
+  - [App options](#app-options)
+- [Error types](#framework-error-types)
+  - [Client errors](#client-errors)
+
+---
+
+## Listener middleware
+Slack apps typically receive and/or respond to one to many incoming events from Slack. This can be something like listening to an Events API event (like when a link associated with your app is shared) or a user invoking one of your app's shortcuts. For each type of incoming request from Slack, there are distinct methods that you can pass **listener middleware functions** to handle and respond to the event.
+
+### Methods
+Below is the current list of methods that accept listener functions. These methods handle specific event types coming from Slack, and typically include an identifing parameter before the listener function. The identifying parameter (included below) narrows the events to specific interactions that your listener function is intended to handle, such as a specific `callback_id`, or a certain substring within a message.
+
+| Method                          | Description |
+|--------------------------------------------------------------------|
+| `app.event(eventType, fn);`     | Listens for Events API events. The `eventType` is a `string` to identify a [specific event](https://api.slack.com/events) to handle (which must be subscribed to in your app's configuration).   | 
```

Comment:
Having a link to https://api.slack.com/events  is a great idea! To avoid confusion, we may want to mention `message.channels` is not an event type.